### DISCUSSION TITLE
Canary training with punctuated/capitalized ASR transcripts is normalizing transcripts to lower case and no special characters

### DISCUSSION THREAD

**Discussion by DeveshS1209** [NVIDIA-NeMo/NeMo#9398] [Q&A] [answered]
So I was trying to train my CANARY model with punctuated data. I have created my training and validation manifest files with punctuated text as shown below:

{"audio_filepath": "audio.wav", "duration": 5.39, "taskname": "asr", "source_lang": "en", "target_lang": "en", "pnc": "yes", "answer": "na", "text": "She is not available that is fine, ok sir can I update this address now ok na."}

However, when I start training my model, all the punctuation and capitalization is gone as shown below:
![image](https://github.com/NVIDIA/NeMo/assets/169297948/2b7e1612-d226-4e91-bc25-085a32897f21)

I am pretty sure that I have removed all the normalization that is happening before putting it to train. I have also trained my tokenizer on punctuated data.

I want to know, if there is any specific implementation in CANARY that is again normalizing my text.

@titu1994

**Comment by MedAymenF** [comment, marked as answer]
Which command have you used to train the tokenizer?
"${NEMO_ROOT}/scripts/tokenizers/process_asr_text_tokenizer.py" applies nmt_nfkc_cf normalization by default.
